packages/cli/src/lifecycles/bundle.js

```diff
@@ -202,84 +206,58 @@ async function bundleStyleResources(compilation, resourcePlugins) {
       const request = new Request(url, { headers });
       const initResponse = new Response(contents, { headers });
 
-      let response = await resourcePlugins.reduce(async (responsePromise, plugin) => {
-        const intermediateResponse = await responsePromise;
+      let response = initResponse;
+
+      for (const plugin in resourcePlugins) {
         const shouldServe = plugin.shouldServe && (await plugin.shouldServe(url, request));
 
         if (shouldServe) {
           const currentResponse = await plugin.serve(url, request);
-          const mergedResponse = mergeResponse(
-            intermediateResponse.clone(),
-            currentResponse.clone(),
-          );
+          const mergedResponse = mergeResponse(response.clone(), currentResponse.clone());
 
           if (mergedResponse.headers.get("Content-Type").indexOf(contentType) >= 0) {
-            return Promise.resolve(mergedResponse.clone());
+            response = mergedResponse.clone();
           }
         }
```

Comment:
Each stage is pretty much the same, so a future refactoring might be to extract each of the stages and let them share some code through a common function. It would also reduce the size of this massive function.